### DISCUSSION TITLE
`dvc pull` not throwing a bash error when it prints "ERROR: unexpected error"

### DISCUSSION THREAD

**Discussion by asmith26** [treeverse/dvc#10400] [Help] [answered]
Hi there,

I'm using DVC in a GitHub Action, and it's not working due to a mismatch of DVC version. I struggled to find this error because `dvc pull` does not seem to be throwing a bash error, so GitHub Action says that step finished correctly.

Is there a way to catch such DVC errors? Here is the current output (I only found this by enabling debug logging for this GitHub Action):

```bash
ERROR: unexpected error

Having any troubles? Hit us up at https://dvc.org/support, we are always happy to help!
```

Many thanks for any help, and this amazing lib!

**Comment by shcheklein** [comment]
@asmith26 can it be related to this? https://github.com/iterative/dvc/issues/10395 ? there is a workaround as well. Please give it a try.

**Reply by asmith26** [reply, marked as answer]
Thanks for the quick response @shcheklein !

I've managed to fix my error (by installing the correct DVC version), it just it took me a long time to find this error because DVC didn't throw a bash error - not sure there is a way to tell `dvc pull` to throw a bash error when it hits a problem (e.g something like this: https://stackoverflow.com/questions/821396/aborting-a-shell-script-if-any-command-returns-a-non-zero-value)?
